automation/src/test/scala/org/broadinstitute/dsde/firecloud/component/Clickable.scala

```diff
@@ -13,10 +23,32 @@ trait Clickable extends LazyLogging { this: Component =>
     * Click on the element modeled by this Component
     */
   def doClick()(implicit webDriver: WebDriver): Unit = {
-    await condition {
-      isVisible && isEnabled
-    }
+    // await condition (isVisible && isEnabled)
+    waitUntilTimeout(ExpectedConditions.elementToBeClickable(query.element.underlying))(_,_)
     scrollToVisible()
-    click on query
+    try {
+      click on query
+    } catch {
+      // No retry when:
+      // 1) WebElement not found
+      // 2)
+      case e: WebDriverException => // retry
+        logger.warn(e.getMessage + s"retry clicking $query")
+        val size: selenium.Dimension = query.element.underlying.getSize()
+        // moveToElement scrolls WebElement into viewport if it was outside viewport
+        new Actions(webDriver).moveToElement(query.element.underlying, -(size.width * 25)/100, 0).click.build.perform()
+    }
+  }
+
+  // helper
+  def waitUntilTimeout[T](cond: => T)(timeout: Int = 30, msg: String)(implicit webDriver: WebDriver): T = {
+
+    val wait = new WebDriverWait(webDriver, timeout)
+    val func = new Function[WebDriver, T]() {
+      override def apply(driver: WebDriver): T = {
+        cond
+      }
+    }
+    wait.withMessage(msg).until(func)
```

Comment:
Is `func` necessary? Shouldn't this just be `wait.withMessage(msg).until(cond)`?